We enable multicast_querier by default.


--- linux-2.6-2.6.32/net/bridge/br_multicast.c.org	2013-03-14 07:16:48.000000000 +0100
+++ linux-2.6-2.6.32/net/bridge/br_multicast.c	2013-03-14 07:17:27.000000000 +0100
@@ -1615,7 +1615,7 @@
 	br->hash_max = 512;
 
 	br->multicast_router = 1;
-	br->multicast_querier = 0;
+	br->multicast_querier = 1;
 	br->multicast_last_member_count = 2;
 	br->multicast_startup_query_count = 2;
 
