--- ixgbe-3.10.16/src/ixgbe_main.c	2012-06-20 22:32:30.000000000 +0200
+++ ixgbe-3.10.16.new/src/ixgbe_main.c	2012-07-19 05:53:51.000000000 +0200
@@ -5194,6 +5194,7 @@
 	struct ixgbe_hw *hw = &adapter->hw;
 	struct net_device *netdev = adapter->netdev;
 	int err;
+	u8 old_addr[ETH_ALEN];
 
 	/* lock SFP init bit to prevent race conditions with the watchdog */
 	while (test_and_set_bit(__IXGBE_IN_SFP_INIT, &adapter->state))
@@ -5228,9 +5229,11 @@
 
 	clear_bit(__IXGBE_IN_SFP_INIT, &adapter->state);
 
+	/* do not flush user set addresses */
+	memcpy(old_addr, &adapter->mac_table[0].addr, netdev->addr_len);
 	ixgbe_flush_sw_mac_table(adapter);
-	memcpy(&adapter->mac_table[0].addr, hw->mac.perm_addr,
-	       netdev->addr_len);
+	memcpy(&adapter->mac_table[0].addr, old_addr, netdev->addr_len);
+
 	adapter->mac_table[0].queue = VMDQ_P(0);
 	adapter->mac_table[0].state = (IXGBE_MAC_STATE_DEFAULT |
 					IXGBE_MAC_STATE_IN_USE);
