Index: openvswitch-2.3.2/acinclude.m4
===================================================================
--- openvswitch-2.3.2.orig/acinclude.m4	2015-06-18 21:32:47.000000000 +0200
+++ openvswitch-2.3.2/acinclude.m4	2015-07-21 10:06:17.000000000 +0200
@@ -291,6 +291,7 @@
   OVS_GREP_IFELSE([$KSRC/include/linux/netdevice.h], [can_checksum_protocol])
   OVS_GREP_IFELSE([$KSRC/include/linux/netdevice.h], [netdev_features_t])
   OVS_GREP_IFELSE([$KSRC/include/linux/netdevice.h], [pcpu_sw_netstats])
+  OVS_GREP_IFELSE([$KSRC/include/net/net_namespace.h], [pernet_operations])
 
   OVS_GREP_IFELSE([$KSRC/include/linux/random.h], [prandom_u32])
 
Index: openvswitch-2.3.2/datapath/linux/compat/net_namespace.c
===================================================================
--- openvswitch-2.3.2.orig/datapath/linux/compat/net_namespace.c	2015-06-18 21:32:47.000000000 +0200
+++ openvswitch-2.3.2/datapath/linux/compat/net_namespace.c	2015-07-21 10:06:17.000000000 +0200
@@ -3,7 +3,7 @@
 #include <net/net_namespace.h>
 #include <net/netns/generic.h>
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,33)
+#ifndef HAVE_PERNET_OPERATIONS
 
 int compat_init_net(struct net *net, struct rpl_pernet_operations *pnet)
 {
Index: openvswitch-2.3.2/datapath/linux/compat/include/net/net_namespace.h
===================================================================
--- openvswitch-2.3.2.orig/datapath/linux/compat/include/net/net_namespace.h	2015-06-18 21:32:47.000000000 +0200
+++ openvswitch-2.3.2/datapath/linux/compat/include/net/net_namespace.h	2015-07-21 10:06:17.000000000 +0200
@@ -3,7 +3,7 @@
 
 #include_next <net/net_namespace.h>
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,33)
+#ifndef HAVE_PERNET_OPERATIONS
 /* for 2.6.32* */
 struct rpl_pernet_operations {
 	int (*init)(struct net *net);
